it/v7/triggerCalculation/TriggerCalculationControllerISpec.scala

```diff
@@ -0,0 +1,228 @@
+          (BAD_REQUEST, "INVALID_REQUEST", INTERNAL_SERVER_ERROR, InternalError),
+          (FORBIDDEN, "NO_SUBMISSION_EXIST", BAD_REQUEST, RuleNoIncomeSubmissionsExistError),
+          (CONFLICT, "CONFLICT", BAD_REQUEST, RuleFinalDeclarationReceivedError),
+          (INTERNAL_SERVER_ERROR, "SERVER_ERROR", INTERNAL_SERVER_ERROR, InternalError),
+          (SERVICE_UNAVAILABLE, "SERVICE_UNAVAILABLE", INTERNAL_SERVER_ERROR, InternalError),
+          (NOT_FOUND, "UNMATCHED_STUB_ERROR", BAD_REQUEST, RuleIncorrectGovTestScenarioError)
+        )
+
+        val extraTysErrors = List(
+          (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+          (BAD_REQUEST, "INVALID_CRYSTALLISE", BAD_REQUEST, FinalDeclarationFormatError),
+          (UNPROCESSABLE_ENTITY, "INVALID_CALCULATION_ID", INTERNAL_SERVER_ERROR, InternalError),
+          (UNPROCESSABLE_ENTITY, "NO_VALID_INCOME_SOURCES", INTERNAL_SERVER_ERROR, InternalError),
+          (UNPROCESSABLE_ENTITY, "NO_SUBMISSIONS_EXIST", BAD_REQUEST, RuleNoIncomeSubmissionsExistError),
+          (UNPROCESSABLE_ENTITY, "CHANGED_INCOME_SOURCES", BAD_REQUEST, RuleIncomeSourcesChangedError),
+          (UNPROCESSABLE_ENTITY, "OUTDATED_SUBMISSION", BAD_REQUEST, RuleRecentSubmissionsExistError),
+          (UNPROCESSABLE_ENTITY, "RESIDENCY_CHANGED", BAD_REQUEST, RuleResidencyChangedError),
+          (UNPROCESSABLE_ENTITY, "ALREADY_DECLARED", BAD_REQUEST, RuleFinalDeclarationReceivedError),
+          (UNPROCESSABLE_ENTITY, "PREMATURE_CRYSTALLISATION", BAD_REQUEST, RuleTaxYearNotEndedError),
+          (UNPROCESSABLE_ENTITY, "CALCULATION_EXISTS", BAD_REQUEST, RuleCalculationInProgressError),
+          (UNPROCESSABLE_ENTITY, "BVR_FAILURE", BAD_REQUEST, RuleBusinessValidationFailureError),
+          (UNPROCESSABLE_ENTITY, "TAX_YEAR_NOT_SUPPORTED", BAD_REQUEST, RuleTaxYearNotSupportedError)
+        )
+
+        (errors ++ extraTysErrors).foreach(args => (serviceErrorTest _).tupled(args))
+      }
+    }
+  }
+
+  private trait Test {
+
+    val nino             = "ZG903729C"
+    val finalDeclaration = true
+
+    val downstreamSuccessBody: JsValue = Json.parse("""
+        |{
+        |  "id" : "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c"
+        |}
+        |""".stripMargin)
+
+    val successBody: JsValue = Json.parse(
+      s"""
+        |{
+        | "calculationId" : "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c"
+        |}
+      """.stripMargin
+    )
+
+    def mtdTaxYear: String
+
+    def downstreamUri: String
+
+    def setupStubs(): StubMapping
+
+    def request(nino: String, taxYear: String, maybeFinalDeclaration: Option[Boolean]): WSRequest = {
+      val suffix = maybeFinalDeclaration.map(d => s"?finalDeclaration=$d").getOrElse("")
+      val uri    = s"/$nino/self-assessment/$taxYear/$suffix"
+
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders(
+          (ACCEPT, "application/vnd.hmrc.5.0+json"),
```

Comment:
Need to update this to use v7.
```suggestion
          (ACCEPT, "application/vnd.hmrc.7.0+json"),
```